packages/rich-text/src/plugins/EmbeddedEntityInline/index.tsx

```diff
@@ -89,24 +90,19 @@ async function selectEntityAndInsert(editor, sdk: FieldExtensionSDK) {
     ...newEntitySelectorConfigFromRichTextField(sdk.field, INLINES.EMBEDDED_ENTRY),
     withCreate: true,
   };
+  const selection = editor.selection;
+
   const entry = await sdk.dialogs.selectSingleEntry<Entry>(config);
+  ReactEditor.focus(editor); // Dialog steals focus from editor, return it.
   if (!entry) return;
 
-  const inlineEntryNode = {
-    type: INLINES.EMBEDDED_ENTRY,
-    children: [{ text: '' }],
-    data: {
-      target: {
-        sys: {
-          id: entry.sys.id,
-          type: 'Link',
-          linkType: 'Entry',
-        },
-      },
-    },
-  };
+  const inlineEntryNode = createInlineEntryNode(entry.sys.id)
 
-  Transforms.insertNodes(editor, inlineEntryNode);
+  // Got to wait until focus is really back on the editor or setSelection() won't work.
+  setTimeout(() => {
```

Comment:
Is it worth using something like [p-wait-for](https://www.npmjs.com/package/p-wait-for) to provide more stability? In other words, is `0` is always enough?

```javascript
waitFor(ReactEditor.isFocused).then(()=> {
    Transforms.setSelection(editor, selection)
    Transforms.insertNodes(editor, inlineEntryNode);
})
```


P.S. if you choose to use `p-wait-for` make sure to stick to `v3` as `v4` uses ESM modules.

Reply:
I believe the problem is that `ReactEditor.isFocused` will be `true` immediately, but I'll double-check.

Reply:
Yes, it's `true` immediately after calling `focus()` :roll_eyes: Also see my [Slack #Slate message](https://slate-js.slack.com/archives/C1RH7AXSS/p1629833275346700?thread_ts=1629790927.323000&cid=C1RH7AXSS).

I'll probably open an issue with them. If `ReactEditor.focus()` would return a promise and `ReactEditor.isFocused()` turns `true` only once the `Promise` resolves, that'd be great.

but there's also the issue of why do we need to call `focus()` in the first place for ` Transforms.setSelection(editor, selection)` to work.